docs/development/buildenvironments.rst

```diff
@@ -80,3 +80,23 @@ DOCKER_VERSION
     Version of the API to use for the Docker API client.
 
     Default: :djangosetting:`DOCKER_VERSION`
+
+Local development
+-----------------
+
+On Linux development environments, it's likely that your UID and GID do not
+match the ``docs`` user that is set up as the default user for builds. In this
+case, it's necessary to make a new image that overrides this user::
+
+    cd contrib/
+    ./docker_build.sh
+
+You can also specify a specific image version to use::
+
+    cd contrib/
+    ./docker_build.sh 4.0rc1
+
+This will create a new image, ``readthedocs/build:dev``, which you can specify
+as the default image in local settings overrides::
+
+    DOCKER_IMAGE = 'readthedocs/build:dev'
```

Comment:
There are some other problems here that worth to have documented.

It's fine defining `DOCKER_IMAGE` here but I found problems when the project defines the docker image in the `readthedocs.yml` file as `latest` for example. In this case, the `readthedocs/build:latest` will be used which is the original image and the build will fail because of the permissions. To fix this, I created a `readthedocs/build:latest_dev` and re-tag it as `readthedocs/build:latest`

    docker tag c59bd4340982 readthedocs/build:latest

Another thing that needs to be changed is `DOCKER_IMAGE_SETTINGS`. My example,

```python
DOCKER_IMAGE_SETTINGS = {
    'readthedocs/build:3.0_dev': {
        'python': {'supported_versions': [2, 2.7, 3, 3.3, 3.5, 3.6]},
    },
    'readthedocs/build:4.0.0rc1_dev': {
        'python': {'supported_versions': [2, 2.7, 3, 3.3, 3.5, 3.6]},
    },
}
```

Reply:
If there is a good way to detect when the dev images exist, this seems like something we can automate for `settings.dev`